app/customer.py

```diff
@@ -0,0 +1,85 @@
+import datetime
+from app.shop import Shop
+from app.car import Car
+from app.point import Point
+
+
+class Customer:
+    def __init__(
+        self,
+        name: str,
+        product_cart: dict,
+        location: Point,
+        money: int | float,
+        car: Car,
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.location = location
+        self.money = money
+        self.car = car
+
+    def check_shops(
+            self, shops: list[Shop],
+            fuel_price: int | float
+    ) -> Shop | None:
+        print(f"{self.name} has {self.money} dollars")
+        shops_costs = {}
+        for shop in shops:
+            distance = self.location.distance_to_point(shop.location)
+            total_cost = round(
+                (
+                    (
+                        shop.price_of_products_set(self.product_cart)
+                        + self.car.cost_of_way(fuel_price, distance) * 2
+                    )
+                ),
+                2,
+            )
+            print(f"{self.name}'s trip to the {shop.name} costs {total_cost}")
+            shops_costs[shop.name] = total_cost
```

Comment:
You can save as key total_cost and by value shop
And that you does not need the loop for minimal shop